homeassistant/components/lupusec/__init__.py

```diff
@@ -39,44 +40,64 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-LUPUSEC_PLATFORMS = [
+PLATFORMS: list[Platform] = [
     Platform.ALARM_CONTROL_PANEL,
     Platform.BINARY_SENSOR,
     Platform.SWITCH,
 ]
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up Lupusec component."""
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the lupusec integration."""
+
+    if DOMAIN not in config:
+        return True
+
     conf = config[DOMAIN]
-    username = conf[CONF_USERNAME]
-    password = conf[CONF_PASSWORD]
-    ip_address = conf[CONF_IP_ADDRESS]
-    name = conf.get(CONF_NAME)
+    conf[CONF_HOST] = conf[CONF_IP_ADDRESS]
```

Comment:
We can remove this one

Reply:
if i remove the line I will get the:

`    host = user_input[CONF_HOST]
           ~~~~~~~~~~^^^^^^^^^^^
KeyError: 'host'`

Error. Since the legacy yaml uses CONF_IP_ADDRESS.

Reply:
I think it's nicest to throw the full config object to the config flow. This way the tests are also covering this and the tests are more representative to the actual yaml config

Reply:
Got it.